includes/Jobs/GenerateProductFeed.php

```diff
@@ -56,13 +77,42 @@ protected function handle_start() {
 		$this->feed_file_handler->write_to_temp_file(
 			$this->feed_data_exporter->generate_header()
 		);
+		// Reset the statistics counters.
+		facebook_for_woocommerce()->get_tracker()->track_feed_file_generation_time( 0 );
+		$this->track_feed_file_generation_start( time() );
+		$this->track_feed_file_generation_time( 0 );
+	}
+
+	/**
+	 * Handle processing a chain batch.
+	 *
+	 * @hooked {plugin_name}/jobs/{job_name}/chain_batch
+	 *
+	 * @param int   $batch_number The batch number for the new batch.
+	 * @param array $args         The args for the job.
+	 *
+	 * @throws Exception On error. The failure will be logged by Action Scheduler and the job chain will stop.
+	 */
+	public function handle_batch_action( int $batch_number, array $args ) {
+		$start_time = microtime( true );
+		parent::handle_batch_action( $batch_number, $args );
+		$generation_time = microtime( true ) - $start_time;
+
+		// Add processing time to previously accumulated execution times.
+		$accumulated_time  = get_option( self::OPTION_FEED_GENERATION_TIME, 0 );
```

Comment:
We can use the _getter_ here instead
```suggestion
		$accumulated_time  = $this->get_feed_file_generation_time();
```